syntax = "proto3";

service Faculdade {
  rpc InserirMatricula (Matricula) returns (Res) {}
  rpc AtualizarNota (Matricula) returns (Res) {}
  rpc AtualizarFaltas (Matricula) returns (Res) {}
  rpc ListarAlunosDaDisciplina (Matricula) returns (ListaAlunos) {}
  rpc ListarBoletimDoAluno (Matricula) returns (Boletim) {}
}

message Res {
  string mensagem = 1;
}

message Tamanho {
  string quantidade = 1;
}

message ListaAlunos {
  repeated Aluno alunos = 1;
}

message Boletim {
  repeated Disciplina disciplina = 1;
  repeated Matricula matricula = 2;
}

message Curso {
  int32 codigo = 1;
  string nome = 2;
}

message Disciplina {
  int32 codigo = 1;
  string nome = 2;
  string professor = 3;
  int32 cod_curso = 4;
}

message Aluno {
  int32 ra = 1;
  string nome = 2;
  int32 periodo = 3;
  int32 cod_curso = 4;
}

message Matricula {
  int32 ra = 1;
  string cod_disciplina = 2;
  int32 ano = 3;
  int32 semestre = 4;
  float nota = 5;
  int32 faltas = 6;
}